src/webcrypto/aes.ts

```diff
@@ -1,33 +1,59 @@
 import { ensureBytes } from '../utils.js';
 import { getWebcryptoSubtle } from './utils.js';
 
-function generate(algo: string, length: number) {
+/**
+ * AAD is only effective on AES-256-GCM or AES-128-GCM. Otherwise it'll be ignored
+ */
+export type Cipher = (
+  key: Uint8Array,
+  nonce: Uint8Array,
+  AAD?: Uint8Array
```

Comment:
I wonder if AAD should be here or in encrypt/decrypt?

Reply:
`aes_256_gcm_siv` is here. But it's more accurate in `encrypt/decrypt`

Reply:
I tried to move it to `encrypt/decrypt` but some tests failed